IntegrationTests/IntegrationScoped.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace IntegrationTests
+{
+    public class IntegrationScoped<TStrategy> where TStrategy : IScopedStrategy, new()
+    {
+
+        protected readonly TStrategy Strategy = new TStrategy();
+
+        [TestInitialize]
+        public void Setup()
+        {
+            Strategy.Setup().Wait();
```

Comment:
In fact is this necessary? can we not just inherit from what ever the implementation is and move the steup tear down to there, thus avoid a layer of indirection?